pypeit/coadd2d.py

```diff
@@ -394,21 +413,32 @@ def reduce(self, pseudo_dict, show=None, show_peaks=None):
         # normally, but this is not possible for coadds
         redux.tilts = pseudo_dict['tilts']
         redux.waveimg = pseudo_dict['waveimg']
+        redux.binning = self.binning
 
         # Masking
-        #  TODO: This is incorrect here JXP and an ugly hack. You need to treat the masking of the slits objects
-        #   from every exposure, come up with an aggregate mask (if it is masked on one slit, mask the slit for all) and that should
-        #   be propagated into the slits object in the psuedo_dict
-        #   Good luck with that JFH.
+        #  TODO: Treat the masking of the slits objects
+        #   from every exposure, come up with an aggregate mask (if it is masked on one slit,
+        #   mask the slit for all) and that should be propagated into the slits object in the psuedo_dict
         slits = self.stack_dict['slits_list'][0]
         reduce_bpm = (slits.mask > 0) & (np.invert(slits.bitmask.flagged(
             slits.mask, flag=slits.bitmask.exclude_for_reducing)))
         redux.reduce_bpm = reduce_bpm
 
         if show:
-            redux.show('image', image=pseudo_dict['imgminsky']*(sciImage.mask == 0), chname = 'imgminsky', slits=True, clear=True)
+            redux.show('image', image=pseudo_dict['imgminsky']*(sciImage.fullmask == 0), chname = 'imgminsky', slits=True, clear=True)
+
         # Object finding
-        sobjs_obj, nobj, skymask_init = redux.find_objects(sciImage.image, show_peaks=show_peaks)
+        # HACK ME!
```

Comment:
I don't think this belongs here. This code should be in reduce so that in general the pipeline can deal with manual extractions.

Reply:
Okay now I see why this has to be here.